sync/sync.go

```diff
@@ -17,17 +17,21 @@ var _ service.Service = (*Synchronizer)(nil)
 
 // Synchronizer manages a list of StarknetData to fetch the latest blockchain updates
 type Synchronizer struct {
-	Blockchain   *blockchain.Blockchain
-	StarknetData starknetdata.StarknetData
+	Blockchain          *blockchain.Blockchain
+	StarknetData        starknetdata.StarknetData
+	StartingBlockNumber *uint64
+	HighestBlockHeader  *core.Header
 
 	log utils.SimpleLogger
 }
 
 func New(bc *blockchain.Blockchain, starkNetData starknetdata.StarknetData, log utils.SimpleLogger) *Synchronizer {
 	return &Synchronizer{
-		Blockchain:   bc,
-		StarknetData: starkNetData,
-		log:          log,
+		Blockchain:          bc,
+		StarknetData:        starkNetData,
+		StartingBlockNumber: nil,
+		HighestBlockHeader:  nil,
```

Comment:
I dont think you need these, they are nil by default anyways.

Reply:
Done